node/network/availability-recovery/src/lib.rs

```diff
@@ -316,6 +482,27 @@ fn reconstructed_data_matches_root(
 	branches.root() == *expected_root
 }
 
+impl Interaction {
+	async fn run(mut self) -> error::Result<()> {
+		loop {
+			match self.phase {
+				InteractionPhase::RequestFromBackers(ref mut from_backers) => {
+					if from_backers.run(&self.params, &mut self.to_state).await? {
```

Comment:
I would suggest to make this a simple `Result` and continue to `RequestChunks` on any error. While the code looks correct as it is now, any accidentally reported error would cause recovery to fail. On the flip side, if we continue to requesting chunks because of a system error, it will just fail as well, so no harm done.

Reply:
I changed things to be a lot more explicit about error handling. This can only fail if the underlying subsystem is unavailable, in which case there is nothing we can do anyway.